gama/genetic_programming/mutation.py

```diff
@@ -106,18 +142,60 @@ def mut_insert(individual: Individual, primitive_set: dict) -> None:
     ----------
     individual: Individual
         Individual to mutate in-place.
-    primitive_set: dict
+    config_space: cs.ConfigurationSpace
+        The ConfigSpace object which defines the search space. Refer to the
+        configuration/(classification||regression).py file for further details.
     """
     parent_node = random.choice(list(individual.primitives))
     new_primitive_node = random_primitive_node(
-        output_type=DATA_TERMINAL, primitive_set=primitive_set
+        output_type=DATA_TERMINAL, config_space=config_space
     )
     new_primitive_node._data_node = parent_node._data_node
     parent_node._data_node = new_primitive_node
 
 
+def has_multiple_options(hyperparameter: cs.hyperparameters.hyperparameter) -> bool:
+    """Check if a ConfigSpace hyperparameter has more than one option.
+
+    Only Constant, Float, Integer, and Categorical hyperparameters are currently
+    supported. A TypeError is thrown if the hyperparameter is not of one of these
+    types. Additionally, readers are directed to our Github Issues page to request
+    support for other types.
+
+    Parameters
+    ----------
+    hyperparameter: cs.hyperparameters.hyperparameter
+        The hyperparameter to check.
+
+    Returns
+    -------
+    bool
+        True if the hyperparameter has more than one option, False otherwise.
+    """
+    if isinstance(
+        hyperparameter,
+        (
+            cs.hyperparameters.FloatHyperparameter,
+            cs.hyperparameters.IntegerHyperparameter,
+        ),
+    ):
+        # For Float and Integer, check if the upper and lower bounds are different
+        return hyperparameter.upper > hyperparameter.lower
```

Comment:
Can you provide an example where this isn't the case? And if this isn't the case, shouldn't be just force it to be defined as a Constant? If not, why not?